### PR TITLE
Tests for api Indexof(string, char, stringcomparision)

### LINKED ISSUE #25094: Proposal: String.Contains(char)
## Rationale and Usage

Finding a character in a string is a fairly common primitive operation. Unfortunately we lead users to the pit of failure because `mystring.Contains(char)` will bind to the Linq version, which can easily be 20x slower than `mystring.IndexOf(char) != -1`. By adding these to string, the next recompile will give a performance improvement.

## Proposed API

```c#
    public sealed partial class String : System.Collections.Generic.IEnumerable<char>, System.Collections.IEnumerable, System.IComparable, System.IComparable<string>, System.IConvertible, System.IEquatable<string>, System.ICloneable
    {
        public bool Contains(char value) { throw null; }
        public bool Contains(char value, StringComparison comparisonType) { throw null; }
        public bool Contains(string value) { throw null; }  // already exists 
        public bool Contains(string value, StringComparison comparisonType) { throw null; } // already exists
        public int IndexOf(char value, StringComparison comparisonType) { throw null; } // to implement above
    }
```

the implementations will simply be

```c#
public bool Contains(char value)
{
    return IndexOf(value) != -1;
}

public bool Contains(char value, StringComparison comparisonType)
{
    return IndexOf(value, comparisonType) != -1;
}
```

## Microbenchmark

Searching in 10 and 1000 character strings:
```
    Method |       Mean |     Error |    StdDev |
---------- |-----------:|----------:|----------:|
 LinqShort | 142.750 ns | 0.4997 ns | 0.4430 ns |
  LinqLong | 180.121 ns | 0.4741 ns | 0.3428 ns |
 FastShort |   7.575 ns | 0.0196 ns | 0.0174 ns |
  FastLong |   8.508 ns | 0.0403 ns | 0.0377 ns |
```

## Variations

The StringComparison overload is in order to search case insensitively if desired. The IndexOf overload is needed to implement it. These could be discarded as the 90% case I would expect to not take a comparison.

### PR DESCRIPTION
**Pull request by Anipik**
Fixes #25094
PR implementation link = https://github.com/dotnet/coreclr/pull/15180

### CODE REVIEW

File: src/System.Runtime/tests/System/StringTests.netcoreapp.cs [hunk 1]
```diff
@@ -577,5 +577,185 @@ public static void ImplicitCast_NullString_ReturnsDefaultSpan()
...
+        public static void IndexOf_TurkishI_EnglishUSCulture_Char()
+        {
+            RemoteInvoke(() =>
+            {
+                CultureInfo.CurrentCulture = new CultureInfo("en-US");
+
+                string s = "Turkish I \u0131s TROUBL\u0130NG!";
+                char value = '\u0130';
+
+                value = '\u0130';
```
Comment (stephentoub) [inline src/System.Runtime/tests/System/StringTests.netcoreapp.cs]: nit: it was just assigned to this on the line above.  You can change the line above to just be `char value;`

**Comment by stephentoub** [comment]
cc: @eerhardt

**Comment by Anipik** [comment]
@danmosemsft should I merge it after the tests ?

**Comment by danmosemsft** [comment]
Yes, you can always merge as soon as
* CI is green
* you have at least one explicit signoff from a Microsoft contributor (green checkmark)
* you addressed all feedback one way or another .. there are no unaddressed comments
* it seems like there aren't any other folks (eg someone with special expertise or area owner) who really ought to look at it 
* it doesn't seem like anyone's in the middle of reviewing it or asked to look at it again

**Pull request merged**